src/UI/Seller/src/app/product-management/components/products-form/products-form.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Product } from '@ordercloud/angular-sdk';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { AppFormErrorService, AppIdValidator } from '@app-seller/shared';
+
+@Component({
+  selector: 'products-form',
+  templateUrl: './products-form.component.html',
+  styleUrls: ['./products-form.component.scss'],
+})
+export class ProductsFormComponent implements OnInit {
+  private _existingProduct: Product = {};
+  @Input() btnText: string;
+  @Output() formSubmitted = new EventEmitter();
+  productForm: FormGroup;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private formErrorService: AppFormErrorService
+  ) {}
+
+  ngOnInit() {
+    this.setForm();
+  }
+
+  @Input()
+  set existingProduct(product: Product) {
+    this._existingProduct = product || {};
+    this.setForm();
+  }
+
+  setForm() {
+    this.productForm = this.formBuilder.group({
+      ID: [this._existingProduct.ID || '', AppIdValidator()],
+      Name: [this._existingProduct.Name || '', Validators.required],
+      Description: [this._existingProduct.Description || ''],
+      Active: [this._existingProduct.Active || false],
+      Featured: [this._existingProduct.xp && this._existingProduct.xp.Featured],
+    });
+  }
+
+  protected onSubmit() {
+    if (this.productForm.status === 'INVALID') {
+      return this.formErrorService.displayFormErrors(this.productForm);
+    }
+
+    const product = {
+      ...this.productForm.value,
+      xp: { Featured: this.productForm.value.Featured },
+    };
+    delete product.Featured;
+
+    this.formSubmitted.emit(product);
```

Comment:
You can remove this delete if you want. Any updates to ordercloud that aren't part of the specified model just don't show up. 

Not a big deal here but could keep the code a little cleaner in the future

Reply:
I'll get rid of it